x-pack/plugins/ml/public/application/datavisualizer/data_comparison/index_patterns_picker.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect, useState, useMemo } from 'react';
+import {
+  EuiPageBody,
+  EuiPageSection,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+} from '@elastic/eui';
+import { parse } from 'query-string';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { SavedObjectFinder } from '@kbn/saved-objects-finder-plugin/public';
+import { type DataViewEditorService as DataViewEditorServiceSpec } from '@kbn/data-view-editor-plugin/public';
+import { INDEX_PATTERN_TYPE } from '@kbn/data-views-plugin/public';
+import { DataDriftIndexPatternsEditor } from './data_drift_index_patterns_editor';
+import { createPath } from '../../routing/router';
+import { ML_PAGES } from '../../../../common/constants/locator';
+
+import { MlPageHeader } from '../../components/page_header';
+import { useMlKibana, useNavigateToPath } from '../../contexts/kibana';
+export const DataDriftIndexOrSearchRedirect: FC = () => {
+  const navigateToPath = useNavigateToPath();
+  const { contentManagement, uiSettings } = useMlKibana().services;
+
+  const nextStepPath = '/data_drift';
+  const onObjectSelection = (id: string, type: string) => {
+    navigateToPath(
+      `${nextStepPath}?${type === 'index-pattern' ? 'index' : 'savedSearchId'}=${encodeURIComponent(
+        id
+      )}`
+    );
+  };
+
+  return (
+    <div data-test-subj="mlPageSourceSelection">
+      <EuiPageBody restrictWidth={1200}>
+        <>
+          <MlPageHeader>
+            <EuiFlexGroup>
+              <EuiFlexItem grow={true}>
+                <FormattedMessage
+                  id="xpack.ml.newJob.wizard.selectDataViewOrSavedSearch"
+                  defaultMessage="Select data view or saved search"
+                />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
```

Comment:
I think it's too easy to miss this button which allows you to create a new (or adhoc) data view. 

<img width="1533" alt="image" src="https://github.com/elastic/kibana/assets/7405507/3b02b6b6-eaf1-4b3a-b4dc-b6c408b0c995">

Could it be moved next to the `Select data view or saved search` heading?

For the text on the control / link, Lens and Discover use `Create a data view`, which is clearer for me, and it would be preferable I think if our labelling here was consistent.

Any thoughts @mdefazio?

Reply:
+1 on label consistency (Create a data view).

I would like to avoid moving the button to the left.  Our page header typically places these buttons on the right (and it's a more straightforward implementation).  

But I do agree with this spanning the whole width that it could be missed.  If we are restricting the width of the body of this page, I would recommend also restricting the width of the header.  This will bring the button closer to the header as you mentioned, but make it feel more connected to the content.

Reply:
Changed it to Create a data view here 

<img width="1725" alt="Screen Shot 2023-09-07 at 10 16 51" src="https://github.com/elastic/kibana/assets/43350163/ba884e51-1b32-45c5-bca7-4ccc6ff4236d">

Reply:
Adding here from our conversation on Slack: 
If possible, let's try and move this to the right of the search bar.  This will align us with the placement of this button on the content management views for Dashboards, Maps, Canvas, etc. I would also suggest matching the styling of the button as well: 

![image](https://github.com/elastic/kibana/assets/3756330/eb3e6228-08ca-46d2-922f-4f71176f667a)


_Example from Dashboard list view:_
<img width="1290" alt="Screenshot--2023-09-08--Dashboards - Elastic" src="https://github.com/elastic/kibana/assets/3756330/8c13eca7-8883-43b3-a14b-144ef349a652">

Reply:
Updated to match with Dashboards here 
[`975992f` (#162853)](https://github.com/elastic/kibana/pull/162853/commits/975992f28f7c0e816394d3b9d7fb2e6cbbe17f3a)


<img width="1713" alt="Screen Shot 2023-09-10 at 11 43 59" src="https://github.com/elastic/kibana/assets/43350163/f240a662-fc81-4aa3-afea-8705f26f1a70">